providers/transip/transipProvider.go

```diff
@@ -114,42 +114,33 @@ func (n *transipProvider) getCorrectionsUsingDiff2(dc *models.DomainConfig, reco
 	}
 
 	for _, change := range instructions {
-		msg := strings.Join(change.Msgs, "\n")
 
-		switch change.Type {
+		if canDirectApplyDNSEntries(change) {
+			changeFunction := n.getChangeFunction(change.Type, dc, change)
+			corrections = append(corrections, change.CreateCorrection(changeFunction))
+		} else {
+			deleteFunction := n.getChangeFunction(diff2.DELETE, dc, change)
+			createFunction := n.getChangeFunction(diff2.CREATE, dc, change)
+			corrections = append(corrections, change.CreateCorrectionWithMessage("[1/2] delete", deleteFunction), change.CreateCorrectionWithMessage("[2/2] create", createFunction))
+		}
+	}
 
-		case diff2.CREATE:
-			corrections = append(corrections, &models.Correction{
-				Msg: msg,
-				F:   wrapChangeFunction(change.New, func(rec domain.DNSEntry) error { return n.domains.AddDNSEntry(dc.Name, rec) }),
-			})
+	return corrections, nil
+}
 
-		case diff2.CHANGE:
-			if canUpdateDNSEntries(change.New, change.Old) {
-				corrections = append(corrections, &models.Correction{
-					Msg: msg,
-					F:   wrapChangeFunction(change.New, func(rec domain.DNSEntry) error { return n.domains.UpdateDNSEntry(dc.Name, rec) }),
-				})
-			} else {
-				corrections = append(corrections, &models.Correction{
-					Msg: fmt.Sprintf("[1/2] delete of %s", msg),
-					F:   wrapChangeFunction(change.Old, func(rec domain.DNSEntry) error { return n.domains.RemoveDNSEntry(dc.Name, rec) }),
-				})
-				corrections = append(corrections, &models.Correction{
-					Msg: fmt.Sprintf("[2/2] create of %s", msg),
-					F:   wrapChangeFunction(change.New, func(rec domain.DNSEntry) error { return n.domains.AddDNSEntry(dc.Name, rec) }),
-				})
-			}
+func (n *transipProvider) getChangeFunction(changeType diff2.Verb, dc *models.DomainConfig, change diff2.Change) func() error {
+	switch changeType {
+	case diff2.DELETE:
+		return wrapChangeFunction(change.Old, func(rec domain.DNSEntry) error { return n.domains.RemoveDNSEntry(dc.Name, rec) })
 
-		case diff2.DELETE:
-			corrections = append(corrections, &models.Correction{
-				Msg: msg,
-				F:   wrapChangeFunction(change.Old, func(rec domain.DNSEntry) error { return n.domains.RemoveDNSEntry(dc.Name, rec) }),
-			})
-		}
+	case diff2.CREATE:
+		return wrapChangeFunction(change.New, func(rec domain.DNSEntry) error { return n.domains.AddDNSEntry(dc.Name, rec) })
+
+	case diff2.CHANGE:
+		return wrapChangeFunction(change.Old, func(rec domain.DNSEntry) error { return n.domains.UpdateDNSEntry(dc.Name, rec) })
 	}
 
-	return corrections, nil
+	panic(fmt.Sprintf("Unsupported change type %s", changeType))
```

Comment:
Suggestion:

    return nil, fmt.Errorf("Unsupported change type %s", changeType)

Reply:
I tried doing it this way, but I opt-ed out of it. Optionally returning an error in this case will make the usage statements on line 119, 121 and 122 really awkward full with conditional statements. And to be honest, the sole reason to have this is, because I need to return (or panic). But this statement should and will not execute ever, not in the "the user made an error" kind of way, no that statement should be statically unreachable at all times. 

From what I can see I can do one of the following:

1. Return an error and make the code  using it full with conditionals to check the errors (which should not happen)
1. Return an error and ignore the error...?
1. Leave it as is and panic (this is my suggested option)
1. Return an empty function
1. Return an change function (which is the least destructive) 
1. or write function without a switch and default to the change function (same as option above)

pick your poison, I will follow

Reply:
Sure changed it to return `nil`.